tests/run-pass/dup-stdout-stderr.rs

```diff
@@ -0,0 +1,18 @@
+// compile-flags: -Zmiri-disable-isolation
+
+#![feature(rustc_private)]
+
+extern crate libc;
+
+fn main() -> std::io::Result<()> {
+    let bytes = b"hello\n";
+    unsafe {
+        let new_stdout = libc::fcntl(1, libc::F_DUPFD, 0);
+        let new_stderr = libc::fcntl(2, libc::F_DUPFD, 0);
+        libc::write(new_stdout, bytes.as_ptr() as *const libc::c_void, 6);
+        libc::write(new_stderr, bytes.as_ptr() as *const libc::c_void, 6);
+    }
+
+    Ok(())
+}
```

Comment:
Nice test. :)

Another one I'd like to see is first closing stdout, then opening a file and dup'ing that to FD 1, and the writing to FD 1 and checking that the output appears in the file.

Reply:
Looks like currently there is no way in miri for FDs 0,1,2 to get reassigned to a file, due to this check:

https://github.com/rust-lang/miri/blob/master/src/shims/posix/fs.rs#L150

Reply:
Do we want to change this behaviour?

Reply:
Yeah, that check should just go away, and `MIN_NORMAL_FILE_FD` be removed. The entire point is to not make FDs 0-2 special.

Reply:
Hmm, that makes sense. But since we are not able to actually close the standard IO streams, perhaps this check can stay as it is for the time being?

Reply:
No, all such checks should be done inside the trait implementations.